core-shaded/pom.xml

```diff
@@ -28,97 +28,243 @@
-                    <resource>META-INF/maven/org.jctools/jctools-core/pom.properties</resource>
-                    <resource>META-INF/maven/org.jctools/jctools-core/pom.xml</resource>
-                  </resources>
-                </transformer>
-                <!-- Pick up the alternate manifest that was extracted from the core module -->
-                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
-                  <resource>META-INF/MANIFEST.MF</resource>
-                  <file>${project.build.directory}/META-INF-shaded/MANIFEST.MF</file>
-                </transformer>
-              </transformers>
-              <!-- Keep the dependencies of driver-core -->
-              <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+              <instructions>
+                <Bundle-SymbolicName>com.datastax.oss.driver.core</Bundle-SymbolicName>
+                <!--
+                Allow importing code from other packages
+                (so reflection-based loading of policies works)
+                -->
+                <DynamicImport-Package>*</DynamicImport-Package>
+                <!--
+                Don't import the packages below because either:
+                1) they were shaded (except Guava which resides in a separate bundle); or
+                2) they aren't OSGi bundles, and the driver bundle can live without them; or
+                2) they are imported by shaded classes, but not used by the driver bundle.
+                -->
+                <Import-Package>
+                  !com.datastax.oss.driver.shaded.netty.*,
+                  !com.datastax.oss.driver.shaded.jctools.*,
+                  !jnr.*,
+                  !net.jcip.annotations.*,
+                  !edu.umd.cs.findbugs.annotations.*,
+                  !com.google.protobuf.*,
+                  !com.jcraft.jzlib.*,
+                  !com.ning.compress.*,
+                  !lzma.sdk.*,
+                  !net.jpountz.xxhash.*,
+                  !org.bouncycastle.*,
+                  !org.conscrypt.*,
+                  !org.apache.commons.logging.*,
+                  !org.apache.log4j.*,
+                  !org.apache.logging.log4j.*,
+                  !org.eclipse.jetty.*,
+                  !org.jboss.marshalling.*,
+                  !sun.misc.*,
+                  !sun.security.*,
+                  *
+                </Import-Package>
+                <!--
+                Export:
+                1) The driver's packages (API and internal);
+                2) All shaded packages, except Guava which resides in a separate bundle.
+                -->
+                <Export-Package>
+                  com.datastax.oss.driver.api.core.*,
+                  com.datastax.oss.driver.internal.core.*,
+                  com.datastax.oss.driver.shaded.netty.*,
+                  com.datastax.oss.driver.shaded.jctools.*
```

Comment:
With today's changes I realized that the previous pattern (`com.datastax.oss.driver.shaded.*`) was too broad; as a consequence, we were _not importing_ but were instead _exporting_ shaded Guava classes. It's a miracle that it worked this way, but now it's fixed (the proper strategy is the opposite: import shaded Guava classes, but not export them since they live in a separate bundle).